ts/features/itwallet/machine/eid/machine.ts

```diff
@@ -133,18 +174,95 @@ export const itwEidIssuanceMachine = setup({
-              target: "#itwEidIssuanceMachine.UserIdentification.Completed",
+              target: "Completed",
               actions: assign(({ event }) => ({
                 identification: { mode: "spid", idpId: event.idp.id }
               }))
             },
             back: {
-              target: "#itwEidIssuanceMachine.UserIdentification.ModeSelection"
+              target: "ModeSelection"
             }
           }
         },
         CiePin: {
-          // TODO
+          entry: [
+            assign(() => ({ cieAuthContext: undefined })), // Reset the CIE context, otherwise retries will use stale data
+            { type: "navigateToCiePinScreen" }
+          ],
+          on: {
+            "cie-pin-entered": [
+              {
+                guard: ({ event }) => event.isNfcEnabled,
+                target: "StartingCieAuthFlow",
+                actions: assign(({ event }) => ({
+                  identification: { mode: "ciePin", pin: event.pin }
+                }))
+              },
+              {
+                target: "ActivateNfc",
+                actions: assign(({ event }) => ({
+                  identification: { mode: "ciePin", pin: event.pin }
+                }))
+              }
+            ],
+            back: {
+              target: "ModeSelection"
+            }
+          }
+        },
+        ActivateNfc: {
+          entry: "navigateToNfcInstructionsScreen",
+          on: {
+            "nfc-enabled": {
+              target: "StartingCieAuthFlow"
+            },
+            back: {
+              target: "CiePin"
+            }
+          }
+        },
+        StartingCieAuthFlow: {
+          description:
+            "Start the preliminary phase of the CIE identification flow.",
+          tags: [ItwTags.Loading],
+          invoke: {
+            src: "startCieAuthFlow",
+            input: ({ context }) => ({
+              walletAttestationContext: context.walletAttestationContext
+            }),
+            onDone: {
+              actions: assign(({ event }) => ({
+                cieAuthContext: event.output
+              })),
+              target: "ReadingCieCard"
+            },
+            onError: {
+              actions: assign(setFailure(IssuanceFailureType.GENERIC)),
+              target: "#itwEidIssuanceMachine.Failure"
+            }
+          }
+        },
+        ReadingCieCard: {
+          description:
+            "Read the CIE card and get back a url to continue the PID issuing flow. This state also handles errors when reading the card.",
+          entry: "navigateToCieReadCardScreen",
+          on: {
+            "cie-identification-completed": {
+              target: "#itwEidIssuanceMachine.UserIdentification.Completed",
+              actions: assign(({ context, event }) => ({
+                cieAuthContext: {
+                  ...context.cieAuthContext!,
+                  callbackUrl: event.url
+                }
```

Comment:
I know it's an edge case and should never happen, but I think its better to not force a non-null assertion. What do you think?

```suggestion
                cieAuthContext: context.cieAuthContext && {
                  ...context.cieAuthContext,
                  callbackUrl: event.url
                }
```